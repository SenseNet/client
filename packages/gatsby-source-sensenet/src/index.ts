export * from './gatsby-node'
export * from './utils'
