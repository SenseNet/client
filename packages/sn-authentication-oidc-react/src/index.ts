import { UserManager, UserManagerSettings, WebStorageStateStore } from 'oidc-client'
export * from './components/authentication-provider'
export * from './components/oidc-secure'
export * from './use-authentication'
export { WebStorageStateStore, UserManagerSettings, UserManager }
