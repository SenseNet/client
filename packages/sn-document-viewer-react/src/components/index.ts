export * from './comment'
export * from './document-widgets'
export * from './page-widgets'
export * from './ConfirmationDialog'
export * from './DocumentViewer'
export * from './DocumentViewerError'
export * from './DocumentViewerLayout'
export * from './DocumentViewerLoading'
export * from './DocumentViewerRegeneratePreviews'
export * from './LayoutAppBar'
export * from './Page'
export * from './PageList'
export * from './styles'
