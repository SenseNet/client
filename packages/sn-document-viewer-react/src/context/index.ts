export * from './api-settings'
export * from './comment-states'
export * from './comments'
export * from './document-data'
export * from './document-permissions'
export * from './localization-context'
export * from './preview-image-data'
export * from './viewer-settings'
export * from './viewer-state'
