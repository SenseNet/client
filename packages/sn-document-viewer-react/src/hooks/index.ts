export * from './use-comment-state'
export * from './use-comments'
export * from './use-document-data'
export * from './use-document-permissions'
export * from './use-document-viewer-api'
export * from './use-localization'
export * from './use-preview-image'
export * from './use-preview-images'
export * from './use-viewer-settings'
export * from './use-viewer-state'
