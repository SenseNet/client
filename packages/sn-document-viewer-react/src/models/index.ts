export * from './DocumentViewerApiSettings'
export * from './Theming'
export * from './viewer-state'
