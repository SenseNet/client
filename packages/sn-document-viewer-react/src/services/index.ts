export * from './image-utils'
export * from './apply-shape-rotations'
