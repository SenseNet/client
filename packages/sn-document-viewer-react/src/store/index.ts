import { Action, applyMiddleware, compose, createStore, Store } from 'redux'
import { ReduxDiMiddleware } from 'redux-di-middleware'
import { DocumentViewerSettings } from '../models'
import { rootReducer, RootReducerType } from './RootReducer'

const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

/**
 * gets a configuration object for the Store instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const getStoreConfig = (settings: DocumentViewerSettings) => {
  const di = new ReduxDiMiddleware(settings.injector)
  di.setInjectable(settings)
  return {
    rootReducer,
    preloadedState: { sensenetDocumentViewer: { documentState: { isLoading: true } } },
    enhancer: composeEnhancers(applyMiddleware(di.getMiddleware())),
  }
}

/**
 * returns a Store object for a Document Viewer instance
 * @param {DocumentViewerSettings} settings The Settings object for the document viewer instance
 */
export const configureStore: (settings: DocumentViewerSettings) => Store<RootReducerType> = settings => {
  const config = getStoreConfig(settings)
  return createStore<RootReducerType, Action, {}, {}>(config.rootReducer, config.preloadedState, config.enhancer)
}

export * from './Comments'
export * from './Document'
export * from './PreviewImages'
export * from './RootReducer'
export * from './Viewer'
