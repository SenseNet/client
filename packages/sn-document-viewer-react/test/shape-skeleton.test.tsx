import { Delete } from '@material-ui/icons'
import { mount } from 'enzyme'
import React from 'react'
import { DocumentPermissionsContext, ShapeSkeleton } from '../src'
describe('ShapesSkeleton component', () => {
  it('should render without crashing', () => {
    const wrapper = mount(
      <ShapeSkeleton
        rotationDegree={0}
        shapeType="redactions"
        zoomRatio={1}
        shape={{
          guid: 'testRedaction',
          imageIndex: 1,
          h: 1,
          w: 1,
          x: 10,
          y: 10,
        }}
        removeShape={() => {}}
        updateShapeData={() => {}}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle keyup/mouseUp events', () => {
    const removeShape = jest.fn()
    const updateShapeData = jest.fn()

    const wrapper = mount(
      <DocumentPermissionsContext.Provider value={{ canEdit: true, canHideRedaction: true, canHideWatermark: true }}>
        <ShapeSkeleton
          rotationDegree={0}
          shapeType="highlights"
          zoomRatio={1}
          shape={{
            h: 100,
            w: 100,
            x: 10,
            y: 10,
            imageIndex: 1,
            guid: '9a324f30-1423-11e9-bcb9-d719ddfb5f43',
          }}
          removeShape={removeShape}
          updateShapeData={updateShapeData}
        />
      </DocumentPermissionsContext.Provider>,
    )

    wrapper.simulate('keyUp', { key: 'Delete' })
    expect(removeShape).toBeCalled()

    wrapper.simulate('keyUp', { key: 'Backspace' })
    expect(removeShape).toBeCalled()
  })

  it('should handle onBlur event', () => {
    const removeShape = jest.fn()
    const updateShapeData = jest.fn()

    const wrapper = mount(
      <DocumentPermissionsContext.Provider value={{ canEdit: true, canHideRedaction: true, canHideWatermark: true }}>
        <ShapeSkeleton
          rotationDegree={0}
          shapeType="annotations"
          zoomRatio={1}
          shape={{
            h: 100,
            w: 100,
            x: 10,
            y: 10,
            text: 'Example Text',
            guid: '9a324f30-1423-11e9-bcb9-d719ddfb5f43',
            lineHeight: 15,
            fontBold: 34,
            imageIndex: 1,
            fontColor: 'red',
            fontFamily: 'arial',
            fontItalic: false,
            fontSize: 12,
          }}
          removeShape={removeShape}
          updateShapeData={updateShapeData}
        />
      </DocumentPermissionsContext.Provider>,
    )
    wrapper.find('#annotation-input').simulate('focus')
    wrapper.find('#annotation-input').simulate('blur')
    expect(updateShapeData).toBeCalled()
  })
})
