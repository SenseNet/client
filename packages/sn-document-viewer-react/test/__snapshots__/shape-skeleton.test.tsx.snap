// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShapesSkeleton component should render without crashing 1`] = `
<ShapeSkeleton
  removeShape={[Function]}
  rotationDegree={0}
  shape={
    Object {
      "guid": "testRedaction",
      "h": 1,
      "imageIndex": 1,
      "w": 1,
      "x": 10,
      "y": 10,
    }
  }
  shapeType="redactions"
  updateShapeData={[Function]}
  zoomRatio={1}
>
  <div
    onBlur={[Function]}
    onClickCapture={[Function]}
    onFocus={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "filter": "",
      }
    }
  >
    <ShapeRedaction
      dimensions={
        Object {
          "height": 1,
          "left": 10,
          "top": 10,
          "width": 1,
        }
      }
      onDragStart={[Function]}
      onResized={[Function]}
      permissions={
        Object {
          "canEdit": false,
          "canHideRedaction": false,
          "canHideWatermark": false,
        }
      }
      removeShape={[Function]}
      shape={
        Object {
          "guid": "testRedaction",
          "h": 1,
          "imageIndex": 1,
          "w": 1,
          "x": 10,
          "y": 10,
        }
      }
      zoomRatio={1}
    >
      <div
        className="makeStyles-root-1 makeStyles-root-3"
        draggable={false}
        key="r-1-1"
        onContextMenu={[Function]}
        onDragStart={[Function]}
        onMouseUp={[Function]}
        tabIndex={0}
      />
      <ForwardRef(Popper)
        anchorEl={null}
        open={false}
        placement="right-start"
      />
    </ShapeRedaction>
  </div>
</ShapeSkeleton>
`;
